apps/chef_objects/src/chef_objects.app.src

```diff
@@ -29,5 +29,71 @@
                   folsom,
                   pooler
                  ]},
-  {env, []}
+  {env, [
+         %% S3 Access credentials for Bookshelf operations
+         {s3_access_key_id, undefined},
+         {s3_secret_key_id, undefined},
+
+         %% s3_url is the endpoint used by the chef-client for
+         %% retreiving cookbooks
+         {s3_url, undefined},
+
+         %% s3_external_url is the endpoint used by Chef Server for
+         %% retreiving cookbooks
+         {s3_external_url, undefined},
```

Comment:
OK, I might have led you astray here, but I think the description of these two is backward.  s3_external_url is the url Chef Server hands back to clients when generating presigned urls.  So it is what chef-client will ultimately see and uses to download files.  s3_url is the url that Chef Server will use internally to make requests to s3 (for instance, when checking for the presence of files via a HEAD request).